File: snippets/s.rb
#!/usr/bin/env ruby

require 'socket'

port_number = 2345
if ARGV.length == 1
  port_number = Integer(ARGV[0])
end

socket = TCPServer.new(port_number)
socket.setsockopt(Socket::SOL_SOCKET, Socket::SO_REUSEADDR, true)

puts "Listening on port #{port_number}"
while true
  client = socket.accept
  client_info = client.peeraddr
  client_host = client_info[2]
  client_port = client_info[1]
  puts "Client: #{client_host}:#{client_port}"
  request = client.gets.chomp
  reply = "s.rb:" + request
  client.puts(reply)
  client.close
end

File: snippets/s.py
#!/usr/bin/env python

import sys, socket

port_number = 2345
if len(sys.argv) == 2:
	port_number = int(sys.argv[1])

#s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
s.bind(('', port_number))
s.listen(1)
print 'Listening on port %d' % port_number
while 1:
	conn, addr = s.accept()
	print type(addr)
	print 'Client: %s:%d' % (addr[0], addr[1])
	request = conn.recv(1024)
	if request:
		print "request: " + request
		reply = "s.py:" + request
		print "reply:   " + reply
		conn.sendall(reply)
		conn.close()
